### PR TITLE
PEMenuCard

### PR DESCRIPTION
**Pull request by Llavrov**

### CODE REVIEW

File: .cspell.json [hunk 1]
```diff
@@ -1,4 +1,4 @@
 {
     "ignorePaths": ["node_modules", "dist", "public"],
-    "words": ["codegen", "esnext", "minifiable", "nextjs", "tailwindcss", "validatorjs"]
+    "words": ["codegen", "esnext", "minifiable", "Carusel", "nextjs", "tailwindcss", "validatorjs"]
```
Comment (CemYil03) [inline .cspell.json:3]: Carousel is written with an O, Carusel -> Carousel. So this spelling mistake should not be ignored by listing it here

File: src/components/cards/menuCard/PEMenuCard.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        pictures: [
+            'https://prykoly.ru/wp-content/uploads/2019/11/Chef9.jpg',
+            'https://prykoly.ru/wp-content/uploads/2019/11/Chef9.jpg',
+            'https://prykoly.ru/wp-content/uploads/2019/11/Chef9.jpg',
+        ],
+        chefPicture: 'https://prykoly.ru/wp-content/uploads/2019/11/Chef9.jpg',
+        categories: ['PEMenuCard', 'PEMealC', 'PEMenuCard'],
+        description: 'PEMenuCard PEMenuCard PEMenuCard PEMenuCard PEMenuCard PEMenuCard PEMenuCard PEMenuCard',
+        title: 'PEMenuCard PEMenuCard PEMenuCard PEMenuCard PEMenuCard PEMea',
+        chefName: 'Loren',
```
Comment (CemYil03) [inline src/components/cards/menuCard/PEMenuCard.stories.tsx:24]: I would suggest to use the property names as mock data, so we know what we are looking at in the storybook.

```suggestion
        categories: ['category-1', 'category-2', 'category-3'],
        description: 'Menu descriptio',
        title: 'Menu title,
        chefName: 'Loren',
```

File: src/components/cards/menuCard/PEMenuCard.tsx [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    onClick?: () => void;
+    title?: string;
+    chefName: string;
+    chefPicture?: string;
+    pricePerPerson?: number;
+    description?: string;
+    pictures?: string[];
+    categories?: string[];
+    kitchens?: string[];
+}
```
Comment (CemYil03) [inline src/components/cards/menuCard/PEMenuCard.tsx:16]: No longer meal card props, but `PEMenuCardProps` and too many thing optional which should not be optional:

```suggestion
export interface PEMenuCardProps {
    onClick?: () => void;
    title: string;
    chefFirstNameName: string;
    chefProfilePictureUrl: string;
    pricePerPerson: number;
    description: string;
    imageUrls: string[];
    categories: string[];
    kitchens: string[];
}
```

File: src/components/cards/menuCard/PEMenuCard.tsx [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    return (
+        <div
+            onClick={onClick}
+            className="flex w-[580px] gap-3 flex-row p-3 box-border rounded-3 shadow-primary cursor-pointer active:shadow-orange hover:shadow-active"
+        >
+            <div className="flex rounded-3 overflow-hidden w-[220px] min-w-[220px] max-w-[220px] h-[220px] max-h-[220px] justify-center items-center bg-base">
+                {pictures && pictures?.length > 1 ? (
+                    <PECarusel
+                        images={pictures?.map((picture) => (
+                            <img key={picture} src={picture} alt={picture} className="w-[220px] h-[220px] object-cover" />
```
Comment (CemYil03) [inline src/components/cards/menuCard/PEMenuCard.tsx:38]: Didn't we say that we want to use the Image component from Next.js?

File: src/components/cards/menuCard/PEMenuCard.tsx [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+                    {kitchens ? (
+                        <div className={'items-center flex flex-row gap-2'}>
+                            <PeopleEatIcon icon={Icon.dishes} />
+                            <div className={'overflow-x-scroll items-center flex flex-row gap-2'}>
+                                {kitchens?.map(
+                                    (menu): ReactElement => (
+                                        <div key={menu} className={'text-orange text-text-s-height '}>
+                                            {menu}
+                                        </div>
+                                    ),
```
Comment (CemYil03) [inline src/components/cards/menuCard/PEMenuCard.tsx:80]: Why do you name each element of an array called `kitchens`, `menu`?
Please rename to kitchen:

```suggestion
                                    (kitchen): ReactElement => (
                                        <div key={kitchen} className={'text-orange text-text-s-height '}>
                                            {kitchen}
                                        </div>
                                    ),
```

File: src/components/cards/menuCard/PEMenuCard.tsx [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+                <div className="flex flex-row justify-between">
+                    <div className="flex flex-row items-center gap-2 overflow-hidden">
+                        {chefPicture ? (
+                            <img src={chefPicture} alt={chefPicture} className="rounded-4 w-6 h-6 object-cover" />
+                        ) : (
+                            <PeopleEatIcon icon={Icon.profileLight} />
+                        )}
+                        <span className="text-preBlack">{chefName}</span>
+                    </div>
+                    {kitchens ? (
```
Comment (CemYil03) [inline src/components/cards/menuCard/PEMenuCard.tsx:71]: Change this to `kitchens && ...`. No ternary operator needed here.

File: src/components/standard/carusel/PECarusel.tsx [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+            //     },
+            //     // Applies to the "prev" button wrapper
+            //     '& .MuiButtonBase-root': {
+            //         '&:hover': {
+            //             backgroundColor: 'rgba(255, 255, 255, 0.2)',
+            //         },
+            //     },
+            // }}
+            // NextIcon={<div className="w-6 h-6 flex justify-center items-center"><PEIcon icon={Icon.arrowNext} edgeLength={8} /></div>}
+            // PrevIcon={<PEIcon icon={Icon.arrowPrev} edgeLength={8} />}
```
Comment (CemYil03) [inline src/components/standard/carusel/PECarusel.tsx:24]: Forgot to remove commented out section?

**Review by CemYil03 — CHANGES_REQUESTED** [review]
In general try to move the `PEMenuCardProps` to out of the component file and reuse it for mobile and desktop component. They will always have the exact same input.

Also, my Feedback for `PEMenuCard` applies almost the same for `PEMenuCardMobile`

File: src/components/cards/menuCard/PEMenuCard.tsx (hunk 4 above)
Comment (Llavrov) [reply src/components/cards/menuCard/PEMenuCard.tsx:38]: Yes, but change it to img temporary to use src like https://prykoly.ru/wp-content/... without errors

File: src/components/standard/carusel/PECarusel.tsx (hunk 7 above)
Comment (Llavrov) [reply src/components/standard/carusel/PECarusel.tsx]: Just keep it to use later for hover effect

File: src/components/cards/menuCard/PEMenuCard.tsx (hunk 4 above)
Comment (CemYil03) [reply src/components/cards/menuCard/PEMenuCard.tsx:38]: Okay, no problem then 👍

File: src/components/cards/mealComponent/PEMealComponent.stories.tsx [hunk 8]
```diff
@@ -10,7 +10,7 @@ export default meta;
 
 export const Component: StoryObj<typeof PEMealComponent> = {
     args: {
-        description: 'PEMealCard PEMealCard PEMealCard PEMealCard PEMealCard PEMealCard PEMealCard PEMealCard',
-        title: 'PEMealCard PEMealCard PEMealCard PEMealCard PEMealCard PEMea',
+        description: 'PEMenuCard PEMenuCard PEMenuCard PEMenuCard PEMenuCard PEMenuCard PEMenuCard PEMenuCard',
+        title: 'PEMenuCard PEMenuCard PEMenuCard PEMenuCard PEMenuCard PEMea',
```
Comment (CemYil03) [inline src/components/cards/mealComponent/PEMealComponent.stories.tsx:14]:
```suggestion
        description: 'Meal description,
        title: 'Meal title',
```

File: src/components/cards/mealComponent/PEMealComponentMobile.stories.tsx [hunk 9]
```diff
@@ -10,7 +10,7 @@ export default meta;
 
 export const Component: StoryObj<typeof PEMealComponentMobile> = {
     args: {
-        description: 'PEMealCard PEMealCard PEMealCard PEMealCard PEMealCard PEMealCard PEMealCard PEMealCard',
-        title: 'PEMealCard PEMealCard PEMealCard PEMealCard PEMealCard PEMea',
+        description: 'PEMenuCard PEMenuCard PEMenuCard PEMenuCard PEMenuCard PEMenuCard PEMenuCard PEMenuCard',
+        title: 'PEMenuCard PEMenuCard PEMenuCard PEMenuCard PEMenuCard PEMea',
```
Comment (CemYil03) [inline src/components/cards/mealComponent/PEMealComponentMobile.stories.tsx:14]:
```suggestion
        description: 'Meal description,
        title: 'Meal title',
```

File: src/components/cards/menuCard/PEMenuCard.tsx [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+    categories,
+    kitchens,
+}: PEMenuCardProps): ReactElement {
+    return (
+        <div
+            onClick={onClick}
+            className="flex w-[580px] gap-3 flex-row p-3 box-border rounded-3 shadow-primary cursor-pointer active:shadow-orange hover:shadow-active"
+        >
+            <div className="flex rounded-3 overflow-hidden w-[220px] min-w-[220px] max-w-[220px] h-[220px] max-h-[220px] justify-center items-center bg-base">
+                {pictures && pictures?.length > 1 ? (
```
Comment (CemYil03) [inline src/components/cards/menuCard/PEMenuCard.tsx:35]: With pictures no longer being optional, we also don't need optional chaining here:

```suggestion
                {pictures && pictures.length > 1 ? (
```

File: src/components/cards/menuCard/PEMenuCard.tsx [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+                )}
+            </div>
+            <div className="flex gap-3 flex-col box-border">
+                <div className="flex flex-col gap-2 h-[148px] overflow-hidden">
+                    <span className="text-text-sm-bold text-preBlack">{title}</span>
+                    {pricePerPerson ? <span className="text-orange text-text-sm-bold">${pricePerPerson} for each person</span> : null}
+                    <span className="text-text-s text-preBlack">{description}</span>
+                </div>
+                <div className={'overflow-x-scroll flex flex-row gap-2'}>
+                    {categories?.map(
```
Comment (CemYil03) [inline src/components/cards/menuCard/PEMenuCard.tsx:54]: Same here:

```suggestion
                    {categories.map(
```

File: src/components/cards/menuCard/PEMenuCard.tsx [hunk 12]
```diff
@@ -0,0 +1,89 @@
...
+                                {kitchens?.map(
+                                    (kitchen): ReactElement => (
+                                        <div key={kitchen} className={'text-orange text-text-s-height '}>
+                                            {kitchen}
+                                        </div>
+                                    ),
+                                )}
+                            </div>
+                        </div>
+                    )}
```
Comment (CemYil03) [inline src/components/cards/menuCard/PEMenuCard.tsx:84]: Kitchens are no longer optional, so we can do the following:

```suggestion

                     <div className={'items-center flex flex-row gap-2'}>
                         <PeopleEatIcon icon={Icon.dishes} />
                         <div className={'overflow-x-scroll items-center flex flex-row gap-2'}>
                             {kitchens.map(
                                 (kitchen): ReactElement => (
                                     <div key={kitchen} className={'text-orange text-text-s-height '}>
                                         {kitchen}
                                     </div>
                                 ),
                             )}
                         </div>
                     </div>
```

File: src/components/cards/menuCard/PEMenuCardMobile.stories.tsx [hunk 13]
```diff
@@ -0,0 +1,20 @@
...
+
+export default meta;
+
+export const Component: StoryObj<typeof PEMenuCardMobile> = {
+    args: {
+        pricePerPerson: 12,
+        kitchens: ['PEMenuCard', 'PEMealC', 'PEMenuCard'],
+        categories: ['PEMenuCard', 'PEMealC', 'PEMenuCard'],
+        description: 'PEMenuCard PEMenuCard PEMenuCard PEMenuCard PEMenuCard PEMenuCard PEMenuCard PEMenuCard',
+        title: 'PEMenuCard PEMenuCard PEMenuCard PEMenuCard PEMenuCard PEMea',
```
Comment (CemYil03) [inline src/components/cards/menuCard/PEMenuCardMobile.stories.tsx:17]:
```suggestion
        kitchens: ['kitchen-1', 'kitchen-2', 'kitchen-3'],
        categories: ['category-1', 'category-2', 'category-3'],
        description: 'Menu description',
        title: 'Menu title',
```

File: src/components/cards/menuCard/PEMenuCardMobile.tsx [hunk 14]
```diff
@@ -1,26 +1,29 @@
...
     title?: string;
+    chefName: string;
     pricePerPerson?: number;
     description?: string;
     picture?: string;
+    chefPicture?: string;
     categories?: string[];
-    menus?: string[];
+    kitchens?: string[];
 }
```
Comment (CemYil03) [inline src/components/cards/menuCard/PEMenuCardMobile.tsx:15]: You can probably delte this and reuse the one form the desktop component

**Review by CemYil03 — CHANGES_REQUESTED** [review]

File: src/components/standard/carusel/PECarusel.tsx (hunk 7 above)
Comment (CemYil03) [reply src/components/standard/carusel/PECarusel.tsx]: Okay 👍

File: src/components/cards/menuCard/PEMenuCard.stories.tsx [hunk 15]
```diff
@@ -0,0 +1,26 @@
...
+    title: 'Cards/PEMenuCard',
+    component: PEMenuCard,
+};
+
+export default meta;
+
+export const Component: StoryObj<typeof PEMenuCard> = {
+    args: {
+        pricePerPerson: 12,
+        kitchens: ['PEMenuCard', 'PEMealC', 'PEMenuCard'],
```
Comment (CemYil03) [inline src/components/cards/menuCard/PEMenuCard.stories.tsx:14]:
```suggestion
        kitchens: ['kitchen-1', 'kitchen-2', 'kitchen-3'],
```

File: src/components/cards/menuCard/PEMenuCard.tsx (hunk 10 above)
Comment (Llavrov) [reply src/components/cards/menuCard/PEMenuCard.tsx:35]: Hm, do you think that pictures must be no optional?

**Review by CemYil03 — APPROVED** [review]

**Pull request merged**